### PR TITLE
Mark next saga events as rejected if SagaTaskSpec#next changed during…

### PR DESCRIPTION
**Pull request by igordumchykov**

### CODE REVIEW

File: src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java [hunk 1]
```diff
@@ -275,19 +338,19 @@ public void successExecuteTaskAndSendNextEvents() {
 
     private SagaEvent mockEvent(String txId, String typeKey, String id) {
         return new SagaEvent().setTenantKey("XM")
-                              .setId(id)
-                              .setTypeKey(typeKey)
-                              .setCreateDate(Instant.now(clock))
-                              .setTransactionId(txId);
+            .setId(id)
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java]: previous alignment better then current

File: src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java [hunk 2]
```diff
@@ -275,19 +338,19 @@ public void successExecuteTaskAndSendNextEvents() {
...
     }
 
     private SagaEvent inQueueEvent(String txId, String typeKey, String id) {
         return new SagaEvent().setTenantKey("XM")
-                              .setId(id)
-                              .setStatus(IN_QUEUE)
-                              .setTypeKey(typeKey)
-                              .setCreateDate(Instant.now(clock))
-                              .setTransactionId(txId);
+            .setId(id)
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java]: previous alignment better then current

**Comment by amedvedchuk** [comment]
@igordumchykov please increment `version` property in the file `gradle.properties`

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java [hunk 3]
```diff
@@ -145,7 +147,10 @@ private void doTask(SagaEvent sagaEvent, Context context) {
             StopWatch stopWatch = StopWatch.createStarted();
             log.info("Start execute task by event {} transaction {}", sagaEvent, transaction);
             Continuation continuation = new Continuation();
+            Set<String> nextTasks =  new HashSet<>(taskSpec.getNext());
             Map<String, Object> taskContext = taskExecutor.executeTask(taskSpec, sagaEvent, transaction, continuation);
+            nextTasks.removeAll(taskSpec.getNext());
+            markAsRejectedByCondition(nextTasks, transaction);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java]: add test that check transaction finished after branch

File: src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java (hunk 1 above)
Comment (igordumchykov) [reply src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java]: Done

File: src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java (hunk 2 above)
Comment (igordumchykov) [reply src/test/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceTest.java]: Done

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java (hunk 3 above)
Comment (igordumchykov) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java]: Done

**Comment by igordumchykov** [comment]
> @igordumchykov please increment `version` property in the file `gradle.properties`

Done

**Pull request merged**